src/net.cpp

```diff
@@ -1995,7 +2004,7 @@ std::vector<AddedNodeInfo> CConnman::GetAddedNodeInfo() const
     }
 
     for (const std::string& strAddNode : lAddresses) {
-        CService service(LookupNumeric(strAddNode, Params().GetDefaultPort(strAddNode)));
+        CService service{MaybeFlipIPv6toCJDNS(LookupNumeric(strAddNode, Params().GetDefaultPort(strAddNode)))};
```

Comment:
The existent code in `master` seems to be comparing the ports. Is there another bug that an inbound connection from `1.2.3.4` would not be considered as "we are connected to `1.2.3.4:8333`"?

Reply:
I mean, if we already have an inbound connection from `1.2.3.4` (ie we are connected to `1.2.3.4:somerandomport`) and this code checks whether we are connected to `1.2.3.4:8333` it would wrongly assume that we are not and would open an outbound connection to that address.